src/main/java/org/sqlite/core/NativeDB.java

```diff
@@ -504,7 +504,9 @@ static String utf8ByteBufferToString(ByteBuffer buffer) {
             return null;
         }
         try {
-            return Charset.forName("UTF-8").decode(buffer).toString();
+            byte[] buff = new byte[buffer.remaining()];
+            buffer.get(buff);
+            return new String(buff, Charset.forName("UTF-8"));
```

Comment:
By the way, we can use StandardCharsets.UTF_8 here https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8

Reply:
Oh, OK.  I think I was following some existing code which was supporting language version < 7 - I'll update.

Reply:
Yeah. I think we don't need to support Java6 anymore.

Reply:
OK, pushed a fix for this.